src/calibre/ebooks/metadata/sources/douban.py

```diff
@@ -170,6 +65,121 @@ class Douban(Source):
     DOUBAN_API_KEY = '0bd1672394eb1ebf2374356abec15c3d'
     DOUBAN_BOOK_URL = 'https://book.douban.com/subject/%s/'
 
+    options = (
+        Option('include_subtitle_in_title', 'bool', True, _('Include subtitle in book title:'),
+               _('Whether to append subtitle in the book title.')),
+    )
+
+    def to_metadata(self, browser, log, entry_, timeout):  # {{{
+        from lxml import etree
+        from calibre.ebooks.chardet import xml_to_unicode
+        from calibre.utils.date import parse_date, utcnow
+        from calibre.utils.cleantext import clean_ascii_chars
+
+        XPath = partial(etree.XPath, namespaces=NAMESPACES)
+        entry          = XPath('//atom:entry')
+        entry_id       = XPath('descendant::atom:id')
+        title          = XPath('descendant::atom:title')
+        description    = XPath('descendant::atom:summary')
+        subtitle       = XPath("descendant::db:attribute[@name='subtitle']")
+        publisher      = XPath("descendant::db:attribute[@name='publisher']")
+        isbn           = XPath("descendant::db:attribute[@name='isbn13']")
+        date           = XPath("descendant::db:attribute[@name='pubdate']")
+        creator        = XPath("descendant::db:attribute[@name='author']")
+        booktag        = XPath("descendant::db:tag/attribute::name")
+        rating         = XPath("descendant::gd:rating/attribute::average")
+        cover_url      = XPath("descendant::atom:link[@rel='image']/attribute::href")
+
+        def get_text(extra, x):
+            try:
+                ans = x(extra)
+                if ans:
+                    ans = ans[0].text
+                    if ans and ans.strip():
+                        return ans.strip()
+            except:
+                log.exception('Programming error:')
+            return None
+
+        id_url = entry_id(entry_)[0].text.replace('http://', 'https://')
+        douban_id = id_url.split('/')[-1]
+        title_ = ': '.join([x.text for x in title(entry_)]).strip()
+        subtitle = ': '.join([x.text for x in subtitle(entry_)]).strip()
```

Comment:
Here is the real changes: to include subtitile in title, if the option is enabled.